### PR TITLE
Add local volume pod affinity tests.

### LINKED ISSUE #56100: More e2e tests for volume scheduling
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind feature
/sig storage
/assign

**What happened**:
Some of these could be integration tests instead of e2e tests.
* Positive + negative tests with StatefulSets and pod affinity/anti-affinity
* Stress test with many pods and many volumes across all nodes
    * It might be interesting to add the static provisioner into the mix for auto-cleanup
* Negative test for multiple PVCs in a pod

### PR DESCRIPTION
**Pull request by cofyc**
**What this PR does / why we need it**:

Add local volume pod affinity tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #56100 (with already merged prs #59682, #60260)

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/persistent_volumes-local.go [hunk 1]
```diff
@@ -1829,7 +1836,24 @@ func createStatefulSet(config *localTestConfig, ssReplicas int32, volumeCount in
 					TopologyKey: "kubernetes.io/hostname",
 				},
 			},
-		},
+		}
+	} else {
+		affinity.PodAffinity = &v1.PodAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
```
Comment (krmayankk) [inline test/e2e/storage/persistent_volumes-local.go]: can we reuse the PodAffinityTerm in both if else cases ?

Comment (cofyc) [reply test/e2e/storage/persistent_volumes-local.go]: sure, done

**Pull request merged**